### ISSUE TITLE
Missing frames in sequence

### ISSUE THREAD

**Issue by anilesec** [facebookresearch/ContactPose#17]
Hi @samarth-robo ,

In the provided sequences(in the dataset), for a few sequences, a couple of frames are missing in between. Is there a reason why they are missing and a way to find out the details about missing frames for all the sequences? 

Eg. seq:  full15_use/bowl/images_full/kinect2_left

This information would be very useful to know :)

Looking forward to hearing from you!

Thanks in Advance!

**Comment by samarth-robo** [comment]
hi @anilesec can you elaborate what you mean by missing frames? You mean annotations are present in `annotations.json` but images are absent?

**Comment by anilesec** [comment]
@samarth-robo No, I mean in some sequences if I visualise them I see a jitter in hand motion. In other words, hand motion is not smooth. For example, in the middle of a sequence hand position jumps drastically from one place to other, which is not smooth looking at overall hand motion of the sequence. In my understanding, this usually happens if the rgb frames in the sequence are either missing or dropped.

**Comment by anilesec** [comment]
For example, for a full1_use/mug/kinnect2_right, I have attached images of frames 165, 166, 167, 168, 169.
Observe the drastic change in hand position from frame 166 to frame 167, which does not seem to be natural.
frame165:
![frame165](https://user-images.githubusercontent.com/23581858/153376190-c17161e5-ef5f-495c-8fa2-58cd7c3b3f8e.png)
frame166:
![frame166](https://user-images.githubusercontent.com/23581858/153376207-87445ef3-dd31-42d1-82aa-8adc1cd2e1ca.png)
frame167:
![frame167](https://user-images.githubusercontent.com/23581858/153376214-633c21c0-a328-494b-aa02-b9f4e78c0caa.png)
frame168:
![frame168](https://user-images.githubusercontent.com/23581858/153376218-9f123010-599e-45a8-8388-02119cf9f53f.png)
frame169:
![frame169](https://user-images.githubusercontent.com/23581858/153376223-4623968e-f779-41a2-9b7d-e8204c97e6e2.png)
 Basically, I am trying to understand the reason behind the drastic change in hand motion between frames 166 and 167

**Comment by samarth-robo** [comment]
@anilesec understood. *Most* of the sequences have consecutive frames with smooth motions, but it is not guaranteed. You can detect discontinuities by looking at the change in position and rotation.

**Comment by anilesec** [comment]
Thanks, @samarth-robo  for confirming. Yes, I can detect the discontinuities but is there any reason for discontinuities? And I am interested in knowing how many frames are missed instead of just detecting discontinuities?

**Comment by samarth-robo** [comment]
@anilesec the reason is that those frames did not pass the RANSAC inlier test for model-fitting 3D joint locations in the object frame. That is described in Section 3.2 of [the paper](https://arxiv.org/pdf/2007.09545.pdf). Code is in a [separate repository](https://github.com/samarth-robo/contactpose_ros_utils).

The number of missing frames varies across each session.
